### DISCUSSION TITLE
Coil uses androidx.appcompat? But Compose doesn't need it?

### DISCUSSION THREAD

**Discussion by ColtonIdle** [coil-kt/coil#834] [Q&A] [answered]
Saw this in the release notes and so I just want some clarity.

A big reason my team is happy about moving to compose is that we were able to officially stop using appcompat. By using coil for compose am I forced to use appcompat, or is this a thing thats only forced on the base coil artifact?

**Comment by colinrtwhite** [comment, marked as answer]
Coil only uses appcompat-resources, though I'd be surprised to hear your app doesn't use androidx.appcompat as it's used by AppCompatActivity.

**Reply by ColtonIdle** [reply]
A compose only app only requires ComponentActivity and therefore we should be able to remove appcompat.

Ian Lake

> AppCompatActivity extends FragmentActivity which extends ComponentActivity. If you don't need AppCompat (i.e. for styling Android Views) and you don't need Fragments (because you're in Compose), then you can use ComponentActivity, which has all of the hooks you need for a Compose only app

> Remove those dependencies (fragment and appcompat) and you might save half a megabyte in APK size 

https://kotlinlang.slack.com/archives/CJLTWPH7S/p1615496894447000?thread_ts=1615494755.443800&cid=CJLTWPH7S

**Reply by colinrtwhite** [reply]
Gotcha, Coil only uses `appcompat-resources` which doesn't import `appcompat`. `appcompat-resources` is a small dependency that only includes backported fixes for drawables + vector drawable support. You can check if appcompat is dependended on by running `./gradlew dependencies`.

**Reply by ColtonIdle** [reply]
Beautiful. Thanks for clarifying!
